internal/provider/kubernetes/controller.go

```diff
@@ -360,6 +370,13 @@ func (r *gatewayAPIReconciler) Reconcile(ctx context.Context, request reconcile.
 		resourceTree.Namespaces = append(resourceTree.Namespaces, namespace)
 	}
 
+	// Add all AuthenticationFilters to the resourceTree.
+	for _, hroute := range resourceTree.HTTPRoutes {
+		if _, ok := resourceMap.httpRouteToAuthenFilters[utils.NamespacedName(hroute)]; ok {
+			resourceTree.AuthenFilters = append(resourceTree.AuthenFilters, resourceTree.AuthenFilters...)
```

Comment:
this is appending the same list, creating a dup of the existing list within it

Reply:
Fixed in [82960d7](https://github.com/envoyproxy/gateway/pull/791/commits/82960d7404475eb7afcc8f059f2fc1127ba1d66e).